### PR TITLE
Implement getting results maps

### PR DESCRIPTION
**Pull request by raimund-schluessler**
This PR implements getting the results data. Will be used in BMicro and can be used in the future for Impose.

Todo:
- [x] Implement correct behaviour of `get_data()` for 3D data
- [x] Write documentation for `get_data()`
- [x] Make bmlab evaluation model backwards compatible (check for existence of `self.parameters` and add it if necessary)
- [x] Finish test functions

### CODE REVIEW

File: bmlab/controllers.py [hunk 1]
```diff
@@ -531,6 +533,76 @@ def extract_payload_spectra(self, image_key):
         evm.set_spectra(image_key, spectra)
         return spectra, times, intensities
 
+    def get_maps(self, parameter_key):
```
Comment (paulmueller) [inline bmlab/controllers.py:536]: The plural in the function name indicates that it returns multiple maps, but in fact it only returns one map, correct?

File: bmlab/controllers.py [hunk 2]
```diff
@@ -531,6 +533,76 @@ def extract_payload_spectra(self, image_key):
...
+        """
+        This function returns the evaluated data,
+        its positions, dimensionality and labels
+        Parameters
+        ----------
+        parameter_key: str
+            The data type requested
```
Comment (paulmueller) [inline bmlab/controllers.py:543]: This is not a "data type", correct? It is a parameter key. How can I find out which parameters are available for an instance of EvaluationController?

**Review by paulmueller — CHANGES_REQUESTED** [review]
Thanks, this looks good to me. I only had one question (hope it's ok to ask this here), one docstring issue and I would change the `get_maps` to `get_map` (If I understood its usage correctly).

File: bmlab/controllers.py (hunk 2 above)
Comment (raimund-schluessler) [reply bmlab/controllers.py:543]: Yes, it's a data key. The available keys can be found via:
https://github.com/BrillouinMicroscopy/BMicro/blob/2a37f7907cbc57e4150e2c0d73aef6b084a1d29a/bmicro/gui/evaluation/evaluation_view.py#L184

File: bmlab/controllers.py (hunk 1 above)
Comment (paulmueller) [reply bmlab/controllers.py:536]: Squeezing is fine if the dimensionality is always returned. I have no preference here. Unsqueezing also does not cost anything.

File: bmlab/controllers.py (hunk 2 above)
Comment (paulmueller) [reply bmlab/controllers.py:543]: OK got it thanks. But please update this doc string here accordingly (maybe even referencing `get_parameter_list`.

**Review by paulmueller — APPROVED** [review]

**Comment by paulmueller** [comment]
Thanks, looking good!

**Pull request merged**